### PR TITLE
Resolve genre modal

### PR DESCRIPTION
**Pull request by lorenzoz23**
i created a lil layer modal to house the genre in to make the overall less cramped, especially on mobile.

i have it so that when you click on genre, it's checked and the modal appears. when you select a genre, the background checkbox updates to show that genre and you can either exit successfully from the modal by clicking outside of it or on the checkmark button. if you click the big x button, the modal will close and the genre checkbox will revert back to its original state.

i didn't add any tests yet or make a new UI component for the genre modal just in case you really don't vibe with it

### CODE REVIEW

**Review by katebrune — CHANGES_REQUESTED** [review]
i think the modal makes sense for mobile, but just leaving the select for web

also, whats the advantage of having the modal on mobile if its the same type of select inside it? maybe use a spinner or something for the select inside the modal ?

edit: like i don't know if it's just because my screen is smaller than yours, but this just feels a mobile component living inside a webpage
image

**Comment by lorenzoz23** [comment]
we don't get to choose the kind of select it is as there is only one with grommet. it is very scrollable so some sort of native ios-type spinner would maybe look better but it would perform the same.

i don't think we should leave the select there for web as i encountered a fair amount of glitchy behavior when it renders which i don't remember being a problem before. also, it gives way to extra space in between the checkboxes above it which leads to a broken symmetry; with the modal, we get what we originally had with the four checkboxes but the convenience of a spinner for selection without taking up anymore space than need be. but i think this will just be something else we'll have jess decide on as we think two different things lol

**Comment by katebrune** [comment]
"we don't get to choose the kind of select it is as there is only one with grommet." -  whats stopping us from adding another library that has the component we want ? why just limit it to grommet ? we already use styled-components, so if its a styling issue, we can do that by hand

"i don't think we should leave the select there for web as i encountered a fair amount of glitchy behavior when it renders" - this is probably css/styling issues that we can fix

i think having the select directly on the screen for web just helps let users know immediately that they're able to select something, instead of hiding it in a modal

but ya, we can let jess decide bc its def a ui-ux decision

**Comment by lorenzoz23** [comment]
i have no problem looking for another select for both web and mobile, but i just never had a problem with the one grommet supplies as both give a seamless experience, especially on mobile for me at least. i also think that that is more of an improvement (finding another picker for mobile) that we can do later. also think that very few selects, even on native apps, are akin to say the one you see in your iPhone's clock app for setting an alarm. most apps i have that have a select, look and feel the same as the one grommet provides. but i'm all about improvements so we can for sure add something to that board for finding a better one for mobile at least.

**Comment by katebrune** [comment]
ayyyy the checkboxes are fixed 🎉 this looks good, last thing i have is can we make the modal less see through? image

File: components/form/form.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { Description } from "../../ui/description/Description"
 import { motion } from "framer-motion"
 import { baseItemBottom } from "../animations/motion"
 import { useSpotify } from "../../common/hooks/useSpotify"
-import { CircularProgress } from "@material-ui/core"
```
Comment (katebrune) [inline components/form/form.tsx:14]: lol we use the material-ui circular progress at work, i really didn't catch & accidentally just used it, that's funny

**Review by katebrune — APPROVED** [review]

**Pull request merged**